ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLChatAgentRunner.java

```diff
@@ -409,6 +424,25 @@ private void runReAct(LLMSpec llm, Map<String, Tool> tools, Map<String, MLToolSp
                     }
                 } else {
                     Object result = output;
+                    MLToolSpec toolSpec = toolSpecMap.get(lastAction.get());
+                    if (toolSpec != null) {
+                        // Runtime level
+                        String includedInAgentResponseKey = toolSpec.getType() + "." + MLToolSpec.INCLUDE_OUTPUT_IN_AGENT_RESPONSE;
+                        boolean includedInAgentResponse = Boolean.parseBoolean(parameters.get(includedInAgentResponseKey));
```

Comment:
We have a parameter `tools` passed in this `runReAct` function. The runtime tool parameters are already overridden when creating the tool in the parent function `runAgent`. So you can have the flag `INCLUDE_OUTPUT_IN_AGENT_RESPONSE` in the tool then use that directly here. No need to combine two parts here.

Reply:
@jngz-es Good suggestion, that might need define a new method in `Tool` interface like below, is that ok we add new method in tool interface?
```java
default boolean includeOutputInAgentResponse() { return false; }
```

And each tool to set the value

Reply:
using toolSpec as only source of this flag, runtime override will be ignored for now